### PR TITLE
feat(opentelemetry): sampling rate configuration option

### PR DESCRIPTION
**Pull request by samugi**
### Summary

Sampling rate can now be configured by plugin (OTel) instead of it just being a global setting
If not configured, the global setting still applies.

This also fixes a small bug where, in the edge case of opentelemetry being used for propagation only (instrumentations disabled), the `sampled` flag was incorrectly set to `true` although no span was sampled for that request.

Includes tests to cover more configuration scenarios (esp. different sampling rates) and verify propagation is done correctly.

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [x] (no) There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* new configuration field for Opentelemetry: `config.sampling_rate` that allows setting sampling_rate "by plugin" instead of globally.

### Issue reference

[KAG-3126](https://konghq.atlassian.net/browse/KAG-3126)

[KAG-3126]: https://konghq.atlassian.net/browse/KAG-3126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by RobSerafini** [comment]
@chobits @brentos - please proved some feedback.

File: kong/pdk/tracing.lua [hunk 1]
```diff
@@ -554,6 +551,51 @@ local function new_tracer(name, options)
...
 
+  --- Get the sampling decision result
+  --
+  -- Uses a parent-based sampler when the parent has sampled flag == false
+  -- to inherit the non-recording decision from the parent span, or when 
+  -- trace_id is not available.
+  --
+  -- Else, apply the probability-based should_sample decision.
+  --
+  -- @function kong.tracing:set_should_sample
```
Comment (brentos) [inline kong/pdk/tracing.lua:562]: Wrong function name here?

File: spec/03-plugins/37-opentelemetry/04-exporter_spec.lua [hunk 2]
```diff
@@ -165,6 +165,87 @@ for _, strategy in helpers.each_strategy() do
...
+          local r = assert(cli:send {
+            method  = "GET",
+            path    = "/",
+            headers = {
+              traceparent = "00-" .. non_sampled_trace_id .. "-0123456789abcdef-01"
+            }
+          })
+          assert.res_status(200, r)
+
+          -- close client connection
```
Comment (chobits) [inline spec/03-plugins/37-opentelemetry/04-exporter_spec.lua:212]: could remove this comment, the source code ` cli:close()` self-explanatory.

**Review by hanshuebner — APPROVED** [review]

File: kong/pdk/tracing.lua (hunk 1 above)
Comment (samugi) [reply kong/pdk/tracing.lua:562]: yep, thanks

**Review by brentos — APPROVED** [review]

**Review by chobits — APPROVED** [review]
